datashape/coretypes.py

```diff
@@ -402,6 +402,67 @@ def to_numpy_dtype(self):
         return np.dtype('O', metadata={'vlen': unicode})
 
 
+class Decimal(Unit):
+    """Decimal type corresponding to SQL Decimal/Numeric types.
+
+    The first parameter passed specifies the number of digits of precision that
+    the Decimal contains. If an additional parameter is given, it represents
+    the scale, or number of digits of precision that are after the decimal
+    point.
+
+    The Decimal type makes no requirement of how it is to be stored in memory,
+    therefore, the number of bytes needed to store a Decimal for a given
+    precision will vary based on the platform where it is used.
+
+    >>> Decimal(18)
+    Decimal(18, 0)
+    >>> Decimal(7,4)
+    Decimal(7, 4)
+    """
+
+    cls = MEASURE
+    __slots__ = 'precision', 'scale'
+
+    def __init__(self, precision, *args):
+        if not args:
+            scale = 0
+        elif len(args) == 1:
+            scale = args[0]
+        else:
+            raise TypeError('Too many arguments passed to Decimal')
+
+        self.precision = precision
+        self.scale = scale
+
+    def __str__(self):
+        return 'decimal[{0}, {1}]'.format(self.precision, self.scale)
+
+    def to_numpy_dtype(self):
+        """
+        Note that floating-point (scale > 0) precision will be lost converting
+        to NumPy floats.
+
+        >>> Decimal(18).to_numpy_dtype()
+        dtype('int64')
+        >>> Decimal(7,4).to_numpy_dtype()
+        dtype('float64')
+        """
+
+        if self.scale == 0:
+            if self.precision <= 2:
+                return np.dtype(np.int8)
+            elif self.precision <= 4:
+                return np.dtype(np.int16)
+            elif self.precision <= 9:
+                return np.dtype(np.int32)
+            elif self.precision <= 18:
+                return np.dtype(np.int64)
+            else:
+                raise TypeError('Integer Decimal precision > 18 is not NumPy-compatible')
+        else:
+            return np.dtype(np.float64)
```

Comment:
That's really a hack due to our reliance on pandas in blaze and odo and the fact that it doesn't have true missing values. We "fake" them by allowing integers with nans to be converted to a float